api/v1/storagecluster_types.go

```diff
@@ -46,6 +46,7 @@ type StorageClusterSpec struct {
 	MonPVCTemplate     *corev1.PersistentVolumeClaim          `json:"monPVCTemplate,omitempty"`
 	MonDataDirHostPath string                                 `json:"monDataDirHostPath,omitempty"`
 	MultiCloudGateway  *MultiCloudGatewaySpec                 `json:"multiCloudGateway,omitempty"`
+	NFS                *NFSSpec                               `json:"nfs,omitempty"`
```

Comment:
it should be part of ManagedResourcesSpec?

Reply:
> it should be part of ManagedResourcesSpec?

I don't think adding NFS to ManagedResourcesSpec and related additional options is really required for tech preview, 
maybe we can add that support later on.

Reply:
AFAIK it should be part of the ManagedResourcesSpec which will provide fine-grained control for the users. For me, it does not feel okay to add the NFS section here its ceph resources like others in the managerResourcesSpec.  @jarrpa @umangachapagain can help here.

Reply:
It seems to me like ManagedResourcesSpec is the right place as well. I see CephBlockPools, CephFileSystems, and CephObjectStores there, and CephNFS is the same form of abstraction as those.

Reply:
According to 
```
	// ManagedResources specifies how to deal with auxiliary resources reconciled
	// with the StorageCluster
	ManagedResources ManagedResourcesSpec `json:"managedResources,omitempty"`
```
This provides extra set of options to controller resources already deployed by OCS-op,
Considering NFS is a new addition, I think we should have it at the top level.
This option will be just to turn it on. 

We can add advanced options later to ManagedResourcesSpec under CephNFSes.